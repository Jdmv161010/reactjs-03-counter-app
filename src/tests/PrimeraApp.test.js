import React from 'react';
import {shallow} from 'enzyme'
import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";

describe('Pruebas en PrimeraApp', () => {
    // test('debe mostrar un mensaje Hola...', () => {
    //     const saludo = 'Hola, soy goku';
    //     const {getByText} = render(<PrimeraApp saludo={saludo}/>);
    //     expect(getByText(saludo)).toBeInTheDocument;
    // })

    test('Debe mostrar PrimeraApp', () => {
        const saludo = 'Hola, soy goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>)

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar subtitulo enviado por props', () => {
        const saludo = 'Hola, soy goku';
        const subtitulo = 'Soy un subtitulo';
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo}/>);

        const textoParrafo = wrapper.find('p').text();
        
        expect(textoParrafo).toBe(subtitulo);
    })
    
    
})
