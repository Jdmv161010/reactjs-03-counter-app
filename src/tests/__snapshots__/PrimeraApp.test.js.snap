// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp Debe mostrar PrimeraApp 1`] = `
<Fragment>
  <h1>
    Hola, soy goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Pruebas en PrimeraApp Debe mostrar subtitulo enviado por props 1`] = `
<Fragment>
  <h1>
    Hola, soy goku
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
